### DISCUSSION TITLE
difference between Context and update in function for send message

### DISCUSSION THREAD

**Discussion by bardiadelagah** [python-telegram-bot/python-telegram-bot#2365] [Q&A]
hi. I have a code like bellow for /start command.
''''''''''''''''''''''''''''''''''''''''''''''''
def start(update, context):
    chat_id = update.message.chat_id
    update.message.reply_text("hi user")
    context.bot.sendMessage(chat_id=chat_id, text='hi user')

start_command = CommandHandler('start', start)
updater.dispatcher.add_handler(start_command)
''''''''''''''''''''''''''''''''''''''''''''''''
I now bot Context and update can send message to user but I can't figuer out what difference between Context and update for send massage to uesr?
 
I search in wiki and doc but can not find any thing

**Comment by Bibo-Joshi** [comment]
`update` is the [`telegram.Update`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.update.html) instance that represents the incoming update from telegram. `context` is an instance of [`telegram.ext.CallbackContext`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.callbackcontext.html) and provides conenience utilities. Note, that `context.bot.send_message(update.message.chat_id, text)` does exactly the same as `update.message.reply_text(text)`, as the latter is just a shortcut for the former.
